airflow/jobs/scheduler_job.py

```diff
@@ -867,122 +631,41 @@ def process_file(
         except Exception:  # pylint: disable=broad-except
             self.log.exception("Failed at reloading the DAG file %s", file_path)
             Stats.incr('dag_file_refresh_error', 1, 1)
-            return [], 0
+            return 0, 0
 
         if len(dagbag.dags) > 0:
             self.log.info("DAG(s) %s retrieved from %s", dagbag.dags.keys(), file_path)
         else:
             self.log.warning("No viable dags retrieved from %s", file_path)
             self.update_import_errors(session, dagbag)
-            return [], len(dagbag.import_errors)
+            return 0, len(dagbag.import_errors)
 
         try:
-            self.execute_on_failure_callbacks(dagbag, failure_callback_requests)
+            self.execute_callbacks(dagbag, callback_requests)
```

Comment:
The `except Exception` here should moved to inside of the execute_callbacks, so that failure in one callback will not drop all other callbacks.

Reply:
Good call, will do

Reply:
Done in https://github.com/apache/airflow/pull/10956/commits/b4764f0b563ad61004bcd73e30b02b160650244c